src/core/hle/kernel/memory.cpp

```diff
@@ -55,14 +75,12 @@ void KernelSystem::MemoryInit(u32 mem_type) {
     }
 
     // We must've allocated the entire FCRAM by the end
-    ASSERT(base == Memory::FCRAM_SIZE);
+    ASSERT(base == is_new_3ds ? Memory::FCRAM_N3DS_SIZE : Memory::FCRAM_SIZE);
```

Comment:
The operation order is likely wrong here. It should be
```suggestion
    ASSERT(base == (is_new_3ds ? Memory::FCRAM_N3DS_SIZE : Memory::FCRAM_SIZE));
```
ASSERT(base == is_new_3ds ? Memory::FCRAM_N3DS_SIZE : Memory::FCRAM_SIZE);